tt_metal/impl/dispatch/memcpy.hpp

```diff
@@ -23,7 +23,6 @@ using vector_memcpy_aligned = std::vector<T, tt::stl::aligned_allocator<T, MEMCP
 template <bool debug_sync = false>
 static inline void memcpy_to_device(void *__restrict dst, const void *__restrict src, size_t n) {
```

Comment:
anybody knows why this code exists? why is it not just `std::memcpy`?

Reply:
For performance 🚀

Reply:
@tt-aho who is the best person to share more about it? Usually such code might be faster in a specific context, sometimes enabling specific compiler optimizations will produce a better result across platforms.

Reply:
do you have any tests which shows that it is faster then memcpy?

Reply:
We have some tests here: `tests/tt_metal/tt_metal/perf_microbenchmark/3_pcie_transfer/test_pull_from_pcie.cpp`

@pgkeller @tt-asaigal for reference regarding usage of avx/non-temporal instructions

Reply:
@tt-aho @pgkeller @tt-asaigal @ayerofieiev-tt @dmakoviichuk-tt  I ran `test_pull_from_pcie` and I observe identical perf values of this code vs `std::memcpy`(I'm using Ryzen 9 7950X3D and a single N150 board).

Let's create a separate ticket to take a look at this, but leave it as-is in this PR :)

Reply:
On the T3Ks we have I think we were seeing ~6GB/s with `std::memcpy`, and ~10GB/s with  `memcpy_to_device`, but we haven't really looked into the perf for this recently. It was a noticeable gain when the custom implementation was originally added.

Reply:
Filed an issue for investigation https://github.com/tenstorrent/tt-metal/issues/15275
Closing this conversation as not related to this PR.